src/Microsoft.Extensions.Options.Validation/OptionsValidationServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Linq.Expressions;
+using Microsoft.Extensions.Options.Validation;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    /// <summary>
+    /// Extension methods for adding options services to the DI container.
+    /// </summary>
+    public static class OptionsValidationServiceCollectionExtensions
+    {
+        /// <summary>
+        /// Adds services required for using options validation.
+        /// </summary>
+        /// <param name="services">The <see cref="IServiceCollection"/> to add the services to.</param>
+        /// <returns>The <see cref="IServiceCollection"/> so that additional calls can be chained.</returns>
+        public static IServiceCollection AddOptionsValidation(this IServiceCollection services)
+        {
+            if (services == null)
+            {
+                throw new ArgumentNullException(nameof(services));
+            }
+
+            services.AddSingleton<IOptionsValidatorManager, OptionsValidatorManager>();
+
+            return services;
+        }
+
+        public static IServiceCollection Validate<TOptions>(this IServiceCollection services, Expression<Func<TOptions, bool>> validateExpression)
```

Comment:
So, you suggest adding a few overloads for `Validate` method with a lightweight signature, correct?

I was thinking about using `Action<TOptions>` as a parameter. But it would require to use exceptions to signal that options are invalid. As far as I remember, using bool is faster than exceptions. Moreover, if will expect exception we will have to wrap into try/catch block each separated validation call (in other case validation will be stopped after first validation failed).

I was trying to use `Func<TOptions, bool>` at first, but I found `Expression` more informative, because it provides information that might help identify the problem (and might be logged).